GW2EIEvtcParser/EncounterLogic/Strikes/SotO/TempleOfFebe.cs

```diff
@@ -333,23 +522,214 @@ private static void AddHiddenWhileNotCasting(NPC target, ParsedEvtcLog log, Comb
+                    }
+                    else
+                    {
+                        lifespanDamageCancelled = ComputeEmbodimentMechanicLifespanWithCancellationTime(log, lifespanDamage);
+                    }
+                    double millisecondsPerDegree = (double)(lifespanDamage.end - lifespanDamage.start) / 360;
+                    double degreesRotated = (lifespanDamageCancelled.end - lifespanDamageCancelled.start) / millisecondsPerDegree;
+                    var rotation2 = new AngleConnector(facing, (float)degreesRotated);
+                    var rectangle2 = (RectangleDecoration)new RectangleDecoration(width, 100, lifespanDamageCancelled, Colors.Red, 0.2, agentConnector).UsingRotationConnector(rotation2);
+                    replay.Decorations.Add(rectangle2);
+                    if (isEmpowered)
+                    {
+                        // Opposite Indicator
+                        var oppositeAgentConnector = (AgentConnector)new AgentConnector(target).WithOffset(new Point3D(-(width / 2), 0), true);
+                        var oppositeRectangle = (RectangleDecoration)new RectangleDecoration(width, 100, lifespanIndicator, Colors.LightOrange, 0.2, oppositeAgentConnector).UsingRotationConnector(rotation);
+                        replay.AddDecorationWithGrowing(oppositeRectangle, growing);
+                        // Opposite Damage
+                        (long start, long end) lifespanDamageOpposite = (lifespanIndicator.end + 950, lifespanIndicator.end + 5850);
+                        (long start, long end) lifespanDamageOppositeCancelled = lifespanDamage;
+                        if (target.IsSpecies(TargetID.Cerus))
+                        {
+                            lifespanDamageOppositeCancelled = ComputeMechanicLifespanWithCancellationTime(log, lifespanDamageOpposite);
+                        }
+                        else
+                        {
+                            lifespanDamageOppositeCancelled = ComputeEmbodimentMechanicLifespanWithCancellationTime(log, lifespanDamageOpposite);
+                        }
+                        double millisecondsPerDegreeOpposite = (double)(lifespanDamageOpposite.end - lifespanDamageOpposite.start) / 360;
+                        double degreedRotatedOpposite = (lifespanDamageOppositeCancelled.end - lifespanDamageOppositeCancelled.start) / millisecondsPerDegreeOpposite;
+                        var rotation3 = new AngleConnector(facing, (float)degreedRotatedOpposite);
+                        var rectangle3 = (RectangleDecoration)new RectangleDecoration(width, 100, lifespanDamageOppositeCancelled, Colors.Red, 0.2, oppositeAgentConnector).UsingRotationConnector(rotation3);
+                        replay.Decorations.Add(rectangle3);
+                    }
+                }
+            }
+        }
+
+        private static (long start, long end) ComputeMechanicLifespanWithCancellationTime(ParsedEvtcLog log, (long start, long end) lifespan)
```

Comment:
This API can already take Cerus as Input as it is always called after a target.IsSpecies(Cerus) check

Reply:
(as an AgentItem, you can then use log.FindActor API)

Reply:
I could merge the 2 into one and have the target.isSpecies(cerus) inside

Reply:
Sounds good to me